trace-utils/src/trace_utils.rs

```diff
@@ -594,9 +596,32 @@ pub fn collect_trace_chunks<T: tracer_payload::TraceChunkProcessor>(
     tracer_header_tags: &TracerHeaderTags,
     process_chunk: &mut T,
     is_agentless: bool,
-) -> TracerPayloadCollection {
+    deduplicate_strings: bool,
```

Comment:
@VianneyRuhlmann Agreed. I think my comment could have been more succulently stated as a rhetorical (and unhelpful) "why didn't we clean this all up earlier?".

Not for this PR, but I also wonder if we should have a concrete type that encapsulates some of the logic present here for a V05 `TracerPayloadCollection` rather than using a tuple.